ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -0,0 +1,241 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.store;
+
+import com.github.ambry.utils.Pair;
+import com.github.ambry.utils.Time;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Exposes stats related to a {@link BlobStore} that is useful to different components.
+ *
+ * Note: This is the v0 implementation of BlobStoreStats. The v0 implementation walks through the entire index
+ * and collect data needed to serve stats related requests.
+ */
+public class BlobStoreStats implements StoreStats {
+  static final String IO_SCHEDULER_JOB_TYPE = "BlobStoreStats";
+  static final String IO_SCHEDULER_JOB_ID = "indexSegment_read";
+
+  private final PersistentIndex index;
+  private final Time time;
+  private final DiskIOScheduler diskIOScheduler;
+
+  BlobStoreStats(PersistentIndex index, Time time, DiskIOScheduler diskIOScheduler) {
+    this.index = index;
+    this.time = time;
+    this.diskIOScheduler = diskIOScheduler;
+  }
+
+  // TODO remove this once #541 is merged
+  public static String getServiceId(StoreKey key) {
+    return key.getID().substring(0, 1);
+  }
+
+  // TODO remove this once #541 is merged
+  public static String getContainerId(StoreKey key) {
+    return key.getID().substring(key.getID().length() - 1);
+  }
+
+  @Override
+  public Pair<Long, Long> getValidDataSize(TimeRange timeRange) throws StoreException {
+    Pair<Long, Map<String, Long>> logSegmentValidSizeResult = getValidDataSizeByLogSegment(timeRange);
+    Long allLogSegmentValidDataSize = 0L;
+    for (Long value : logSegmentValidSizeResult.getSecond().values()) {
+      allLogSegmentValidDataSize += value;
+    }
+    return new Pair<>(logSegmentValidSizeResult.getFirst(), allLogSegmentValidDataSize);
+  }
+
+  @Override
+  public Pair<Long, Map<String, Long>> getValidDataSizeByLogSegment(TimeRange timeRange) throws StoreException {
+    long referenceTimeInSecs = timeRange.getEndTimeInSecs();
+    Map<String, Long> logSegmentValidSizeMap = collectValidDataSizeByLogSegment(referenceTimeInSecs * Time.MsPerSec);
+    return new Pair<>(referenceTimeInSecs, logSegmentValidSizeMap);
+  }
+
+  @Override
+  public Map<String, Map<String, Long>> getValidDataSize() throws StoreException {
+    long referenceTimeInMs = time.milliseconds();
+    return collectValidDataSizeByContainer(referenceTimeInMs);
+  }
+
+  /**
+   * Walk through the entire index and collect valid data size information per container.
+   * @param referenceTimeInMs the reference time to be used to decide whether or not a blob is valid
+   * @return a nested {@link Map} of serviceId to containerId to valid data size
+   */
+  private Map<String, Map<String, Long>> collectValidDataSizeByContainer(long referenceTimeInMs)
```

Comment:
this method collects data not just for container, but also for serviceIds. In fact serviceId is the higher level construct. Try to think if we can come up with a better name. If not, we can leave it as is